backend/src/main/java/ai/giskard/service/ProjectService.java

```diff
@@ -93,7 +123,124 @@ public void delete(Long id) {
         } catch (Exception e) {
             throw new GiskardRuntimeException("Failed to delete project %s".formatted(project.getKey()));
         }
+    }
+
+    public byte[] export(Long id) throws IOException {
+        Project project = projectRepository.getById(id);
+        String projectKey = project.getKey();
+        Path homeProject = locationService.resolvedProjectHome(project.getKey());
+
+        Files.createDirectories(locationService.resolvedTmpPath());
+        Files.createDirectories(locationService.metadataDirectory(projectKey));
+
+        // Conversion To YAML
+        Path projectMetadataPath =  locationService.resolvedMetadataPath(projectKey, Project.class);
+        Path modelsMetadatPath = locationService.resolvedMetadataPath(projectKey, ProjectModel.class);
+        Path datasetsMetadataPath = locationService.resolvedMetadataPath(projectKey, Dataset.class);
+        Path feedBacksMetadataPath = locationService.resolvedMetadataPath(projectKey, Feedback.class);
+        Path testSuitesMetadataPath = locationService.resolvedMetadataPath(projectKey, TestSuite.class);
+
+        YAMLConverter.exportEntityToYAML(project, projectMetadataPath);
+        YAMLConverter.exportEntitiesToYAML(project.getModels(), modelsMetadatPath);
+        YAMLConverter.exportEntitiesToYAML(project.getDatasets(), datasetsMetadataPath);
+        YAMLConverter.exportEntitiesToYAML(project.getFeedbacks(), feedBacksMetadataPath);
+        YAMLConverter.exportEntitiesToYAML(project.getTestSuites(), testSuitesMetadataPath);
+
+        // Zip process
+        Path zipPath = locationService.resolvedTmpPath().resolve(String.format("%s.zip", project.getKey()));
+        if (!Files.exists(zipPath)){
+            Files.createFile(zipPath);
+        }
+        ZipUtils.zip(homeProject, zipPath);
+
+        return Files.readAllBytes(zipPath);
+    }
+
+
+    public Project importProject(MultipartFile zipMultipartFile, String userName) throws IOException {
+        Path pathToTmp = locationService.resolvedTmpPath();
+        String projectKey;
+        try{
+            projectKey = zipMultipartFile.getOriginalFilename().split("[.]")[0];
```

Comment:
it's best to read project key from a project.yaml since users may rename the zip file itself